addons/van_jsx/src/jsx-internal.d.ts

```diff
@@ -5,15 +5,18 @@ import { FunctionChild } from "./type";
 type OriginalElement = HTMLElement;
 
 export declare namespace JSX {
-  type HTMLAttributes<T> = Partial<Omit<T, "style" | "children">> & {
-    style?: CSS.Properties;
-    ref?: State<T>;
-    children?: ChildDom;
+  type JSXProp<T, K extends keyof T> = K extends "children"
```

Comment:
It seems a ts utility type. It can be placed outside the JSX namespace

Reply:
Omit is more readable than long ternary expressions.

Reply:
For some reason when a dynamic key type is used adding additional properties doesn't work. 
Maybe this can be solved by using an intersection type: 
```ts
type HTMLAttributes<T> = { [K in keyof Omit<T, "style" | "children">]: JSXProps<T, K> } & { children: ChildDom, style: CSS.Properties };
```
Will implement this way then 👍🏼

Reply:
Should be resolved now @cqh963852